### PR TITLE
Fix relations/properties pointing to document folders

### PR DESCRIPTION
**Pull request by blankse**
### Bug

When a relation, an object/document property or an editable relation **targets a document folder**, the GraphQL query fails with:

```
unknown document type: document_folder   (GraphQL/Service.php, getDocumentTypeDefinition)
```

The `resolveType()` methods of the relation/property resolvers handle the `document` element type via `getDocumentTypeDefinition('document_' . $documentType)`. For a folder, $documentType is `folder` → `document_folder`, which is not a registered type (only `_document_folder` is). This makes any query that can resolve a document folder through such a field error out.

### Fix

Add a folder guard that returns the registered `_document_folder` type — mirroring how object folders are already handled (`_object_folder`).

Affected resolvers:
- `src/GraphQL/DataObjectType/AbstractRelationsType.php`
- `src/GraphQL/General/AnyTargetType.php`
- `src/GraphQL/General/AnyDocumentTargetType.php`
- `src/GraphQL/PropertyType/ObjectsType.php`

### Notes

Analogous to the object-folder fix. `php -l` clean on all four files.

### Related

Complementary to #1126. This PR only adds the `resolveType()` folder guard (so a folder value resolves to `_document_folder`). For that resolved type to be a valid member of the relation/property unions, it also needs #1126, which makes the `document` union folder-inclusive. Both are required for relations/properties targeting a document folder — please merge together.

### CODE REVIEW

File: src/GraphQL/DataObjectType/AbstractRelationsType.php [hunk 1]
```diff
@@ -140,6 +140,10 @@ public function resolveType($element, $context, ResolveInfo $info): ObjectType|s
                 if ($document) {
                     $documentType = $document->getType();
                     $service = $this->getGraphQlService();
+                    if ($documentType === 'folder') {
+                        return $service->getDocumentTypeDefinition('_document_folder');
+                    }
```
[the remark this answers is not in the document]
Comment (blankse) [reply src/GraphQL/DataObjectType/AbstractRelationsType.php:145]: Good catch. Fixed in 6b8034b1: `_document_folder` is now added to `getTypes()` in the `getDocumentsAllowed()` block, so the resolved runtime type is a valid member of the union.

File: src/GraphQL/PropertyType/ObjectsType.php [hunk 2]
```diff
@@ -85,6 +85,10 @@ public function resolveType($element, $context, ResolveInfo $info): ObjectType|s
                 if ($document) {
                     $documentType = $document->getType();
                     $service = $this->getGraphQlService();
+                    if ($documentType === 'folder') {
+                        return $service->getDocumentTypeDefinition('_document_folder');
+                    }
```
[the remark this answers is not in the document]
Comment (blankse) [reply src/GraphQL/PropertyType/ObjectsType.php:90]: Good catch. Fixed in 6b8034b1: `_document_folder` is now registered in `getTypes()` under the `querySchemaEnabled('document_folder')` guard, mirroring `AnyTargetType`.

File: src/Resources/public/studio/build/987f95c1-e4bf-4f6a-bccd-0df0e850a2d8/entrypoints.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        "/bundles/pimcoredatahub/studio/build/987f95c1-e4bf-4f6a-bccd-0df0e850a2d8/static/js/async/__federation_expose_default_export.87553f32.js",
+        "/bundles/pimcoredatahub/studio/build/987f95c1-e4bf-4f6a-bccd-0df0e850a2d8/static/js/async/__federation_expose_plugins.c4cf01bd.js",
+        "/bundles/pimcoredatahub/studio/build/987f95c1-e4bf-4f6a-bccd-0df0e850a2d8/static/js/remoteEntry.js"
+      ],
+      "css": []
+    },
+    "exposeRemote": {
+      "js": [
+        "/bundles/pimcoredatahub/studio/build/987f95c1-e4bf-4f6a-bccd-0df0e850a2d8/exposeRemote.js"
+      ],
```
[the remark this answers is not in the document]
Comment (blankse) [reply src/Resources/public/studio/build/987f95c1-e4bf-4f6a-bccd-0df0e850a2d8/entrypoints.json]: Obsolete: the duplicate build directory `987f95c1-…` was removed by a subsequent "Automatic frontend build" commit. Only a single build/entrypoint set (`04ab6b52-…`) remains on the branch.

File: src/GraphQL/DataObjectType/AbstractRelationsType.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (blankse) [reply src/GraphQL/DataObjectType/AbstractRelationsType.php:145]: Update: I reverted the explicit `getTypes()` registration again. `_document_folder` is instead advertised as a possible type by making the `document` union itself folder-inclusive (`DocumentType::getTypes()`), which this relation union already merges — so the resolved type stays a valid union member without being listed twice. This PR is now scoped to the `resolveType()` folder guard only.

File: src/GraphQL/PropertyType/ObjectsType.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (blankse) [reply src/GraphQL/PropertyType/ObjectsType.php:90]: Update: I reverted the explicit `getTypes()` registration again. `_document_folder` is advertised via the folder-inclusive `document` union that this union merges, avoiding a duplicate union member. This PR is now scoped to the `resolveType()` folder guard only.

**Comment by robertSt7** [comment]
@blankse Could you please add a regression test for this fix? `tests/GraphQL/ResolveTest.php` doesn't currently cover relation/property/editable resolution, so nothing would catch this case. Ideally it'd test resolving a relation/property pointing at a document folder, together with #1126 since both are needed for it to actually work end-to-end.

**Comment by astapc** [comment]
@blankse ping

**Comment by blankse** [comment]
Added in f1b92056: `tests/GraphQL/DocumentFolderResolveTypeTest.php`.

It creates a real `Document\Folder` and a `Document\Page` (via `TestHelper`) and asserts `resolveType()` on all four resolvers this PR touches — `AbstractRelationsType` (through `HrefType`), `PropertyType\ObjectsType`, `AnyTargetType` and `AnyDocumentTargetType` — returns the `_document_folder` type for the folder and still `document_page` for the page. Without the fix the folder case looks up `document_folder` and throws "unknown document type", so the test fails on the old code.

Two notes:

- I added `Service::getDocumentDataTypes()` as the counterpart of the existing `getDataObjectDataTypes()`, so the test can restore the registered document types after temporarily registering its own instead of reaching into the object with reflection (SonarCloud flags that as `php:S3011`). Say the word if you would rather not widen the API and I will find another way.
- The end-to-end half you mentioned really does need #1126: this PR only makes `resolveType()` return the folder type, whereas the union has to *contain* `_document_folder` for a query to succeed — that is `DocumentType::getTypes()` in #1126. So a test that runs an actual query cannot be green on this branch alone. I can add the union-membership test (`DocumentType::getTypes()` contains `_document_folder`, no duplicate members in `AnyTargetType`/`AnyDocumentTargetType`) to #1126, and if you want a real query-level test once both are merged, I would add it as a follow-up on the target branch. Just tell me which you prefer.

**Comment by robertSt7** [comment]
@blankse Thanks for fixing this pointing
